### PR TITLE
rgw: remove dout_subsys defs from header files

### PR DESCRIPTION
**Pull request by ivancich**
Each compilation unit should be able to define its own dout_subsys
without generating a redefinition warning. When dout_subsys is defined
in header files, it complicates this matter. This commit removes
definitions and header files and makes sure definitions are added to
.cc files as needed.

Signed-off-by: J. Eric Ivancich <ivancich@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by adamemerson — APPROVED** [review]
This is a good PR.

File: src/rgw/rgw_sync_trace.cc [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 
 #define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
```
Comment (adamemerson) [inline src/rgw/rgw_sync_trace.cc:19]: I'd be inclined to make this a `static constexpr` but that's just nitpicking.

File: src/rgw/store/dbstore/common/dbstore_log.h [hunk 2]
```diff
@@ -12,7 +12,6 @@
 #include <fstream>
 #include "common/dout.h"
 
-#define dout_subsys ceph_subsys_rgw
 #undef dout_prefix
 #define dout_prefix *_dout << "rgw dbstore: "
```
Comment (adamemerson) [inline src/rgw/store/dbstore/common/dbstore_log.h:16]: Ideally we should probably not define `dout_prefix` in a header file, either, but I think the only way around that would be making a DoutPrefixProvider.

File: src/rgw/rgw_sync_trace.cc (hunk 1 above)
Comment (ivancich) [reply src/rgw/rgw_sync_trace.cc:17]: Nice idea. So I did that with the files I needed to modify and a few other RGW .cc files as a reminder to do it more broadly.

**Comment by ivancich** [comment]
Because this only affects compilation and not execution, teuthology tests are not requied.

**Comment by ivancich** [comment]
Capturing CI checks when I reverted my one commit so it matched that "main" snapshot.

Screen Shot 2022-08-30 at 1 02 32 PM

**Comment by cbodley** [comment]
jenkins test make check

**Pull request merged**